targets/tinyfpga_bx/Makefile.mk

```diff
@@ -7,6 +7,8 @@ endif
 # Settings
 DEFAULT_TARGET = base
 TARGET ?= $(DEFAULT_TARGET)
+DEFAULT_FIRMWARE = micropython
```

Comment:
I think we want a `DEFAULT_CPU_VARIANT`?

Reply:
Aside from the problem that it's too late to set `CPU` related defaults in the `TARGET` Makefile (the [include of the `TARGET` Makefile](https://github.com/timvideos/litex-buildenv/blob/master/Makefile#L115) happens *after* [the `CPU` / `CPU_VARIANT` settings](https://github.com/timvideos/litex-buildenv/blob/7fc049aad997d8deefb5980fc559fce5f2170675/Makefile#L57-L84)), there's also the problem that if we could set `DEFAULT_CPU_VARIANT` then `minimal` no longer appears in the prompt (since the `CPU_VARIANT` is suppressed from the prompt if `CPU_VARIANT = DEFAULT_CPU_VARIANT`), and I think it's useful that it appears in the prompt at least in this case (as it's a really cut down CPU which is *slow*).

Besides, *not* setting `DEFAULT_CPU_VARIANT` is bug-for-bug compatible with what happens before.

Ewen

Reply:
Can we just move the CPU stuff to after the `TARGET` Makefile?

Reply:
*Maybe* :-)

I did think about that, and in general I think it's a good idea.  But I certainly don't have time right now to audit whether there's anything that currently relies on CPU being set before the TARGET Makefile.  (And the `target/tinyfpga_bx/Makefile.mk` now relies on that... :-)  But that one is at least easily fixable by setting a default CPU there too.)

So this work around seemed like a less invasive change.   Feel free to look into the larger refactoring.... :-)

Ewen